src/codegen/dead_storage.rs

```diff
@@ -580,31 +604,43 @@ fn get_storage_definition<'a>(
     def: &InstrDef,
     cfg: &'a ControlFlowGraph,
 ) -> Option<(usize, &'a Expression)> {
-    match &cfg.blocks[def.block_no].instr[def.instr_no] {
-        Instr::LoadStorage { storage, res, .. } => Some((*res, storage)),
-        _ => None,
+    if let InstrDef::Instr { block_no, instr_no } = def {
+        match &cfg.blocks[*block_no].instr[*instr_no] {
+            Instr::LoadStorage { storage, res, .. } => Some((*res, storage)),
+            _ => None,
+        }
+    } else {
+        None
     }
 }
 
 fn get_definition<'a>(def: &InstrDef, cfg: &'a ControlFlowGraph) -> Option<&'a Expression> {
-    match &cfg.blocks[def.block_no].instr[def.instr_no] {
-        Instr::LoadStorage { storage, .. } => Some(storage),
-        Instr::Set { expr, .. } => Some(expr),
-        _ => None,
+    if let InstrDef::Instr { block_no, instr_no } = def {
+        match &cfg.blocks[*block_no].instr[*instr_no] {
+            Instr::LoadStorage { storage, .. } => Some(storage),
+            Instr::Set { expr, .. } => Some(expr),
+            _ => None,
+        }
+    } else {
+        None
     }
 }
 
 fn get_vars_at(def: &InstrDef, block_vars: &BlockVars) -> ReachingDefs {
-    let vars = if let Some(vars) = block_vars.get(&def.block_no) {
-        vars[def.instr_no].clone()
-    } else {
-        ReachingDefs {
-            vars: HashMap::new(),
-            stores: Vec::new(),
-        }
-    };
+    if let InstrDef::Instr { block_no, instr_no } = def {
+        let vars = if let Some(vars) = block_vars.get(block_no) {
+            vars[*instr_no].clone()
+        } else {
+            ReachingDefs {
+                vars: HashMap::new(),
+                stores: Vec::new(),
+            }
+        };
 
-    vars
+        vars
+    } else {
+        unreachable!();
```

Comment:
Please, leave a message inside `unreachable`, explaining why no flow should read this line.

Reply:
That is a good point. I've added some words,

Reply:
Why not `unreachable!("description")`?

Reply:
Another good point. Updated.